services/navigation/builtin/builtin.go

```diff
@@ -553,16 +554,7 @@ func (svc *builtIn) startWaypointMode(ctx context.Context, extra map[string]inte
 					svc.logger.Infof("skipping waypoint %+v since it was deleted", wp)
 					continue
 				}
-
-				svc.logger.Infof("skipping waypoint %+v due to error while navigating towards it: %s", wp, err)
-				if err := svc.waypointReached(ctx); err != nil {
-					if svc.waypointIsDeleted() {
-						svc.logger.Infof("skipping waypoint %+v since it was deleted", wp)
-						continue
-					}
-					svc.logger.Infof("can't mark waypoint %+v as reached, exiting navigation due to error: %s", wp, err)
-					return
-				}
+				svc.logger.Infof("retrying navigation to waypoint %+v since it errored out: %s", wp, err)
```

Comment:
@raybjork @biotinker 

outline of my logic for setting it up this way.
If move on globe returns an error we check if the waypoint has been deleted.
If it has been deleted we move to the next one, otherwise we retry.

Reply:
I think as written this will skip it since the call to NextWaypoint happens at the top of the loop